### PR TITLE
Autodetect build directory ignoring mkosi artefacts

### PR DESCRIPTION
**Pull request by keszybz**
@poettering ?

### CODE REVIEW

File: test/run-integration-tests.sh [hunk 1]
```diff
@@ -1,21 +1,19 @@
...
 
 RESULT=0
 FAILURES=0
 
+cd "$(dirname "$0")"
 for TEST in TEST-??-* ; do
         echo -e "\n--x-- Starting $TEST --x--"
         set +e
-        make -C "$TEST" BUILD_DIR=$(pwd)/../build clean setup run
+        make -C "$TEST" "BUILD_DIR=$BUILD_DIR" clean setup run
```
Comment (poettering) [inline test/run-integration-tests.sh]: i figure BUILD_DIR=$BUILD_DIR can be dropped entirely, no?

Comment (keszybz) [reply test/run-integration-tests.sh]: Not really. BUILD_DIR is not exported in run-integration-tests.sh. We *could* export it, but I think it's nicer to pass it explicitly. Then it shown up in the logs, and it's also easier to c&p the command for testing.

**Comment by keszybz** [comment]
`KVM: entry failed, hardware error 0x0`

**Comment by evverx** [comment]
This PR seems to have broken the use case mentioned in `README.testsuite` saying that it is possible to do something like this:
```
$ cd test/TEST-01-BASIC
$ sudo make clean setup run
```
. The CI doesn't fail, but it is most likely because `BUILD_DIR` is specified explicitly.

**Comment by poettering** [comment]
> This PR seems to have broken the use case mentioned in README.testsuite saying that it is possible to do something like this:
> 
> $ cd test/TEST-01-BASIC
> $ sudo make clean setup run
> 
> . The CI doesn't fail, but it is most likely because BUILD_DIR is specified explicitly.

hmm, fails here too in a very weird way. @evverx can you file an issue about this?

**Comment by poettering** [comment]
Ah, i figure it's quicker if i just do that. done in #7648.

**Pull request merged**